compiler/optimizer/src/main/java/edu/snu/nemo/compiler/optimizer/pass/compiletime/annotating/UpfrontCloningPass.java

```diff
@@ -20,11 +20,12 @@
 import edu.snu.nemo.common.ir.edge.executionproperty.CommunicationPatternProperty;
 import edu.snu.nemo.common.ir.vertex.IRVertex;
 import edu.snu.nemo.common.ir.vertex.executionproperty.ClonedSchedulingProperty;
+import edu.snu.nemo.compiler.optimizer.pass.compiletime.Requires;
 
 /**
  * Set the ClonedScheduling property of source vertices, in an upfront manner.
  */
-@Annotates(ClonedSchedulingProperty.class)
+@Requires(ClonedSchedulingProperty.class)
```

Comment:
It requires `CommunicationPatternProperty` and annotates `ClonedSchedulingProperty`